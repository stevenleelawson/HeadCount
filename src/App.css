/*APP COMPONENT RELATED CSS GOES HERE*/
.app {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-direction: column;
}
