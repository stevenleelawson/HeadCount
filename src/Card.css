html {
  font-size: 16px;
}
.above {
  color: #0582af;
}

.below {
  color: #af3b05;
}

ul {
  list-style-type: none;
}

.card {
  border: 1px solid black;
  width: 18rem;
  margin: .5rem;
}

.district-data{
  pointer-events: none;
}

.district-name{
  pointer-events: none;
}

.selected{
  background-color: red;
}