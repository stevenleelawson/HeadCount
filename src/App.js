import React, { Component } from 'react';
import './App.css';
import KinderData from './data/kindergartners_in_full_day_program';
import DistrictRepository from './helper';
import CardContainer from './CardContainer';
import CompareCard from './CompareCard';
import Header from './Header';


class App extends Component {
  constructor() {
    super();
    this.state = {
      districtArray: [],
      selectedCards: [],
      averages: null
    };

    this.districts = new DistrictRepository(KinderData);
  }

  retrieveData = (userInput = '') => {
    const schools = this.districts.findAllMatches(userInput);
    this.setState({ districtArray: schools});
  }

  handleClick = (event) => {
    const location = event.target.id;
    const selectedCard = this.districts.findByName(location);
    const selectedArray = this.state.selectedCards;
    const alreadyThere = selectedArray.some(district => district.location === location);

    if (alreadyThere) {
      const selectedCards = selectedArray.filter((district) => district.location !== selectedCard.location);
      this.setState({selectedCards});
      return;
    }

    if (selectedArray.length === 0) {
      this.setState({ selectedCards : [selectedCard] });
    } else if (selectedArray.length === 1) {
      this.setState({ selectedCards : [...selectedArray, selectedCard] });
      this.getAverageOfSelected(selectedArray[0], selectedCard);
    } else if (selectedArray.length === 2) {
      const newSelected = [...selectedArray];
      newSelected.shift();
      this.setState({ selectedCards : [...newSelected, selectedCard] });
      this.getAverageOfSelected(newSelected[0], selectedCard);
    }

  }

  getAverageOfSelected = (location1, location2) => {
    const comparison = this.districts.compareDistrictAverages(location1.location, location2.location);
    this.setState({averages:comparison});
  }

  componentDidMount() {
    this.retrieveData();
  }

  render() {
    return (
      <div className='app'>
        <Header filterSchools={this.retrieveData}/>
        {
          this.state.averages &&
          <CompareCard averages={this.state.averages}/>
        }
        <CardContainer schools={this.state.districtArray} handleClick={this.handleClick} toggleClass={this.toggleClass} selected={this.state.selectedCards}/>

      </div>
    );
  }
}

export default App;
