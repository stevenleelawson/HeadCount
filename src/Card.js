import React from 'react';
import './Card.css'
import PropTypes from 'prop-types';

const Card = (props) => {
  const dataMap = Object.keys(props.data).map( year => {
    return props.data[year] > 0.5 ? <li className="above">{year}: {props.data[year]}</li>: <li className="below">{year}: {props.data[year]}</li> })
  return (
    <div className= {`card ${props.selectedClass}`} onClick={props.handleClick} id={props.location}>
      <h1 className="district-name">{props.location}</h1>
      <ul className="district-data">{dataMap}</ul>
    </div>
  )
}

Card.propTypes = {
  location: PropTypes.string.isRequired,
  data: PropTypes.object.isRequired,
  handleClick: PropTypes.func.isRequired,
  selectedClass: PropTypes.string.isRequired
}

export default Card;
