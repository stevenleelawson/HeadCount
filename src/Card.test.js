import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import PropTypes from 'prop-types';
import  Card  from './Card';



describe('Card', () => {

  it('should match the snapshot', () => {
    const data = {location: 'Denver', data: {'2006': .85 }};
    const wrapper = shallow(<Card data={data}/>);
    expect(wrapper).toMatchSnapshot();
  });

});
