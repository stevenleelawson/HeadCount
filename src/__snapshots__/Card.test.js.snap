// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Card
    data={
        Object {
            "data": Object {
              "2006": 0.85,
            },
            "location": "Denver",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1
          className="district-name"
/>,
        <ul
          className="district-data"
>
          <li
                    className="below"
          >
                    location
                    : 
                    Denver
          </li>
          <li
                    className="below"
          >
                    data
                    : 
                    Object {
                              "2006": 0.85,
                            }
          </li>
</ul>,
      ],
      "className": "card undefined",
      "id": undefined,
      "onClick": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "district-name",
        },
        "ref": null,
        "rendered": null,
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <li
              className="below"
>
              location
              : 
              Denver
</li>,
            <li
              className="below"
>
              data
              : 
              Object {
                            "2006": 0.85,
                          }
</li>,
          ],
          "className": "district-data",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "location",
                ": ",
                "Denver",
              ],
              "className": "below",
            },
            "ref": null,
            "rendered": Array [
              "location",
              ": ",
              "Denver",
            ],
            "type": "li",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "data",
                ": ",
                Object {
                  "2006": 0.85,
                },
              ],
              "className": "below",
            },
            "ref": null,
            "rendered": Array [
              "data",
              ": ",
              Object {
                "2006": 0.85,
              },
            ],
            "type": "li",
          },
        ],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="district-name"
/>,
          <ul
            className="district-data"
>
            <li
                        className="below"
            >
                        location
                        : 
                        Denver
            </li>
            <li
                        className="below"
            >
                        data
                        : 
                        Object {
                                    "2006": 0.85,
                                  }
            </li>
</ul>,
        ],
        "className": "card undefined",
        "id": undefined,
        "onClick": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "district-name",
          },
          "ref": null,
          "rendered": null,
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <li
                className="below"
>
                location
                : 
                Denver
</li>,
              <li
                className="below"
>
                data
                : 
                Object {
                                "2006": 0.85,
                              }
</li>,
            ],
            "className": "district-data",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "location",
                  ": ",
                  "Denver",
                ],
                "className": "below",
              },
              "ref": null,
              "rendered": Array [
                "location",
                ": ",
                "Denver",
              ],
              "type": "li",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "data",
                  ": ",
                  Object {
                    "2006": 0.85,
                  },
                ],
                "className": "below",
              },
              "ref": null,
              "rendered": Array [
                "data",
                ": ",
                Object {
                  "2006": 0.85,
                },
              ],
              "type": "li",
            },
          ],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
